fuel-block-producer/src/ports.rs

```diff
@@ -0,0 +1,25 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use fuel_core_interfaces::{
+    common::fuel_tx::CheckedTransaction,
+    model::{
+        BlockHeight,
+        DaBlockHeight,
+    },
+};
+use std::sync::Arc;
+
+#[async_trait]
+pub trait Relayer: Sync + Send {
+    /// Get the best finalized height from the DA layer
+    async fn get_best_finalized_da_height(&self) -> Result<DaBlockHeight>;
```

Comment:
I guess it also should be sync=)

Reply:
Yeah this could be sync, since we making a bunch of changes to the relayer right now anyways I'll just make this change since we won't be stuck with using a `block_on`.

Reply:
fixed